### PR TITLE
Add Rust to env

### PR DESCRIPTION
**Pull request by jasonodoom**
#  #

## 🗣 Description ##

This PR adds Rust to our env so it can be consumed/used in GitHub Actions workflows.  Specifically, [dtolnay/rust-toolchain](https://github.com/dtolnay/rust-toolchain)

## 💭 Motivation and context ##

This change is based on current team practices. 

## 🧪 Testing ##

Pre-commit verified syntactical accuracy. 

## ✅ Pre-approval checklist ##

- [X] This PR has an informative and human-readable title.
- [X] Changes are limited to a single goal - *eschew scope creep!*
- [X] All relevant type-of-change labels have been added.
- [X] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [X] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [X] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Comment by dav3r** [comment]
Can you please give us more context about why Rust is needed?  Do we have some pending work or PR that will utilize it?  This info should be included in the "Motivation and context" section above.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -33,6 +33,8 @@ jobs:
           # it relies on the existence of a go.sum file.
           cache: false
           go-version: "1.20"
+      - id: setup-rust
+        uses: dtolnay/rust-toolchain@stable
```
Comment (jsf9k) [inline .github/workflows/build.yml:37]: Why not use [the more official GH Action](https://github.com/marketplace/actions/rust-toolchain)?

**Comment by jsf9k** [comment]
> Can you please give us more context about why Rust is needed? Do we have some pending work or PR that will utilize it? This info should be included in the "Motivation and context" section above.

I took [@mcdonnnj's comment](https://github.com/cisagov/skeleton-generic/pull/143#issuecomment-1696083973) and [the successful linting action](https://github.com/cisagov/skeleton-generic/actions/runs/6001655442/job/16276320573?pr=143) in cisagov/skeleton-generic#143 to indicate that this is unnecessary.

File: .github/workflows/build.yml (hunk 1 above)
Comment (mcdonnnj) [reply .github/workflows/build.yml:37]: Looking at the repo it's very dead. The last release was in 2020.

**Comment by mcdonnnj** [comment]
> > Can you please give us more context about why Rust is needed? Do we have some pending work or PR that will utilize it? This info should be included in the "Motivation and context" section above.
> 
> I took [@mcdonnnj's comment](https://github.com/cisagov/skeleton-generic/pull/143#issuecomment-1696083973) and [the successful linting action](https://github.com/cisagov/skeleton-generic/actions/runs/6001655442/job/16276320573?pr=143) in [cisagov/skeleton-generic#143](https://github.com/cisagov/skeleton-generic/pull/143) to indicate that this is unnecessary.

Adding on to this per the [runner image software list](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#rust-tools) the Rust tool-chain should already be available.

**Comment by jasonodoom** [comment]
> > > Can you please give us more context about why Rust is needed? Do we have some pending work or PR that will utilize it? This info should be included in the "Motivation and context" section above.
> > 
> > 
> > I took [@mcdonnnj's comment](https://github.com/cisagov/skeleton-generic/pull/143#issuecomment-1696083973) and [the successful linting action](https://github.com/cisagov/skeleton-generic/actions/runs/6001655442/job/16276320573?pr=143) in [cisagov/skeleton-generic#143](https://github.com/cisagov/skeleton-generic/pull/143) to indicate that this is unnecessary.
> 
> Adding on to this per the [runner image software list](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#rust-tools) the Rust tool-chain should already be available.

@jsf9k @mcdonnnj Aah, I didn't know this. In that case, closing.

**Pull request closed without merging**